app/services/strategy_marketplace_service.py

```
@@ -1756,18 +1769,76 @@ async def _get_user_strategy_portfolio_impl(self, user_id: str) -> Dict[str, Any
+            if redis_client:
+                await self._safe_redis_operation(redis_client.delete, redis_key)
+                for strategy_id in valid_strategy_ids:
+                    await self._safe_redis_operation(redis_client.sadd, redis_key, strategy_id)
+
+            self.logger.info(
+                "Hydrated user strategies from database",
+                user_id=user_id,
+                strategy_count=len(valid_strategy_ids)
+            )
+
+            return valid_strategy_ids
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)